src/components/LeafletCluster.tsx

```diff
@@ -0,0 +1,29 @@
+import L, { Marker } from "leaflet";
+import "leaflet.markercluster/dist/leaflet.markercluster";
+import "leaflet.markercluster/dist/MarkerCluster.css";
+import "leaflet.markercluster/dist/MarkerCluster.Default.css";
+import React, { useEffect } from "react";
+import { useLeaflet } from "react-leaflet";
+
+const markerClusterGroup = L.markerClusterGroup({
+  disableClusteringAtZoom: 17,
+  spiderfyOnMaxZoom: false
+});
+
+type Props = {
+  markers: Marker[];
+};
+
+export const LeafletCluster: React.FC<Props> = ({ markers }) => {
+  const { map } = useLeaflet();
+
+  useEffect(() => {
+    markerClusterGroup.clearLayers();
+
+    markers.map(marker => marker.addTo(markerClusterGroup));
+
+    map?.addLayer(markerClusterGroup);
```

Comment:
Shouldn't there be something like `map.removeLayer` to clean-up on change?

I think this will cause memory leaks as layers are only added, not removed 🤔

Reply:
No. The same layer (markerClusterGroup) but with different markers is added to the map.
I checked all map layers IDs and there are no old layers except unchanged layers.

You can check it yourself by code below to the src/components/LeafletCluster.tsx on line line 26:```
map?.eachLayer(function(layer) {
  // @ts-ignore
  console.log(layer._leaflet_id);
});
```